docs/python-examples/fixtures/test_fixtures.py

```diff
@@ -0,0 +1,134 @@
+import logging
+
+import pytest
+
+logger = logging.getLogger(__name__)
+
+
+def alter_pvc(pvc):
+    """
+    Dummy helper function to alter pvc
+
+    Args:
+        pvc (PVC): PVC object reference
+    """
+    logger.info(f"Doing something with pvc: {pvc}")
+
+
+@pytest.fixture(scope='class')
```

Comment:
I will add some commit on top of this to cleanup some examples like this which are confusing

Reply:
This one looks ok. It can be useful to see how to created basic setup fixture providing more pregenerated resources. On the other hand, developer can look into some test in `tests` directory and look for some examples there.

Reply:
This specific case was discussed on automation mtg AFAIR and that's why I added it here, so don't think its confusing, or can you elaborate what is confusing here? Please before sending additional commit to this PR, please raise the comment about what is confusing or why and how it should be changed to have some discussion before putting effort to changing stuff. I see that this is already approved by @andrewschoen and @clacroix12 . Anyway I see some suggestions from @fbalak which I will fix tomorrow.

Reply:
we have pvc_factory that is an example for creating "N" objects, so we have to  Also, I touched about this topic in our last meeting, we wanted to document in simple terms for QE folks on how they can follow new fixture methods and not get into details of how classes are designed(it was never discussed afaik)  and how delete is handled for them, 

Also,  is_delete flag is unnecessary, since its a class level feature(outside the scope of finalizer) and whether you check in finalizer or not it doesn't matter.

I have pushed my commit on top, we can discuss on monday and this I also want to document in gdoc. This has *wider* implications for all team members, So I want to ensure everybody is getting the core concept of what we are saying here.

Reply:
Here is my diff: https://github.com/red-hat-storage/ocs-ci/pull/428/commits/c731450a2872cee90a11843d574a2775d872b05b